Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
I agree the `toOptional` sounds like it's inverse but do you think that calling `optional` on `Decoded<T>` would return `Decoded<T?>` ... I don't think I would infer that from the name. I don't have a better idea though so I'm ok with it if everyone else is.